quickwit/quickwit-metastore/src/metastore/index_metadata.rs

```diff
@@ -171,6 +171,17 @@ impl IndexMetadata {
         Ok(())
     }
 
+    pub(crate) fn toggle_source(&mut self, source_id: &str, enable: bool) -> MetastoreResult<()> {
```

Comment:
Let's return `MetastoreResult<bool>` and return whether the source was mutated or not.

Reply:
NVM, the metastore is not as smart as I thought.

Reply:
I added it because some like file metastore account for mutation or not. We can improve the other later.